tests/test_config.py

```
@@ -791,5 +799,168 @@ def test_no_env_vars_returns_empty_dict(self):
+    @patch("json.load", side_effect=json.JSONDecodeError("Invalid JSON", "", 0))
+    def test_load_credentials_invalid_json(
+        self, mock_json_load, mock_open, mock_exists
+    ):
+        """Test credential loading with invalid JSON."""
+        result = load_credentials()
+        self.assertIsNone(result)
+
+    @patch("os.path.exists", return_value=True)
+    @patch("builtins.open", new_callable=mock_open)
+    @patch("json.load")
+    def test_load_credentials_success(self, mock_json_load, mock_open, mock_exists):
```

Ruff (0.13.3)

Unused method argument: `mock_open`

(ARG002)